### PR TITLE
chore: use stricter TS config for 3id and blockchain-related packages

### PR DESCRIPTION
**Pull request by ukstv**

### CODE REVIEW

File: packages/3id-did-resolver/src/index.ts [hunk 1]
```diff
@@ -109,9 +115,9 @@ function extractMetadata(
   const { timestamp: updated, cid: versionId } = lastAnchorOrGenesisEntry(requestedVersionState.log)
 
   const { timestamp: nextUpdate, cid: nextVersionId } =
-    latestVersionState.log.find(
-      ({ timestamp }) => timestamp > updated || (!updated && timestamp)
-    ) || {}
+    latestVersionState.log.find(({ timestamp }) => {
+      return (timestamp && updated && timestamp > updated) || (!updated && timestamp)
```
Comment (dbcfd) [inline packages/3id-did-resolver/src/index.ts:119]: It's _ok_ like this, but might be more readable as `return (timestamp && ((updated && timestamp > updated) || !updated))`

Comment (dbcfd) [reply packages/3id-did-resolver/src/index.ts:119]: As it is currently written, it just takes a little bit longer to understand.

File: packages/3id-did-resolver/src/legacyResolver.ts [hunk 2]
```diff
@@ -63,14 +63,22 @@ const compressKey = (key: string) => {
...
+  didId: string,
+  ipfs = ipfsMock
+): Promise<LegacyResolverResult> {
   const doc = (await ipfs.dag.get(didId)).value
   let signingKey, encryptionKey
 
   try {
-    const keyEntrySigning = doc.publicKey.findIndex((e) => e.id.endsWith('signingKey'))
-    const keyEntryEncryption = doc.publicKey.findIndex((e) => e.id.endsWith('encryptionKey'))
+    const keyEntrySigning = doc.publicKey.findIndex((e: any) => e.id.endsWith('signingKey'))
```
Comment (dbcfd) [inline packages/3id-did-resolver/src/legacyResolver.ts:80]: This seems like it should have a type other than any.

File: packages/blockchain-utils-linking/src/ethereum.ts [hunk 3]
```diff
@@ -133,16 +133,16 @@ function safeSend(provider: any, method: string, params?: Array<any>): Promise<a
 
   if (provider.request) {
     return provider.request({ method, params }).then(
-      (response) => response,
-      (error) => {
+      (response: any) => response,
```
Comment (dbcfd) [inline packages/blockchain-utils-linking/src/ethereum.ts:136]: Same as above, be nice if this had a type other than any.

**Review by dbcfd — CHANGES_REQUESTED** [review]

File: packages/3id-did-resolver/src/index.ts (hunk 1 above)
Comment (ukstv) [reply packages/3id-did-resolver/src/index.ts]: Yup. Changed that to the suggested expression.

File: packages/blockchain-utils-linking/src/ethereum.ts (hunk 3 above)
Comment (ukstv) [reply packages/blockchain-utils-linking/src/ethereum.ts]: I just realised that `.then` call here is a no-op :/ so removed it.

**Review by dbcfd — APPROVED** [review]

**Pull request merged**